src/components/App/MainMenu/MainMenu.tsx

```diff
@@ -303,104 +203,113 @@ const StyledNav = styled.nav`
-  ].filter(Boolean)
-
-  const { containerRef } = useFocusTrap({
-    disabled: !isMenuButtonVisible || !isOpen,
-  })
-
-  return (
-    <StyledNav
-      className={classList.join(' ')}
-      ref={containerRef as React.RefObject<HTMLElement>}
-    >
-      {homeLink}
-
-      <div className="claim">{claim}</div>
-
-      <div id="main-menu" role="menu">
-        <AppLinks />
-      </div>
-
-      {closeButton}
-    </StyledNav>
-  )
+	const [isMenuButtonVisible, setIsMenuButtonVisible] = React.useState(false);
+
+	const onResize = React.useCallback(() => {
+		if (window.innerWidth > MENU_BUTTON_VISIBILITY_BREAKPOINT) {
+			setIsMenuButtonVisible(false);
+		} else {
+			setIsMenuButtonVisible(true);
+			props.onToggle(false);
+		}
+	}, [props.onToggle, setIsMenuButtonVisible]);
+
+	React.useEffect(() => {
+		window.addEventListener("resize", onResize);
+		onResize();
+		return () => {
+			window.removeEventListener("resize", onResize);
+		};
+	}, []);
+
+	const toggleMenu = React.useCallback(
+		(event: React.MouseEvent<HTMLButtonElement>) => {
+			props.onToggle(!props.isOpen);
+			event.preventDefault();
+		},
+		[props.onToggle, props.isOpen],
+	);
+
+	const productName =
+		translatedStringFromObject(
+			props.clientSideConfiguration.textContent?.product.name,
+		) || "Wheelmap";
+
+	const homeLink = (
+		<div className="home-link">
+			<Link href="/" legacyBehavior>
+				<Button
+					className="home-Button"
+					aria-label={t`Home`}
+					variant="ghost"
+					radius="none"
+					color="blue"
+				>
+					<VectorImage
+						className="logo"
+						svg={props.clientSideConfiguration.branding?.vectorLogoSVG}
+						aria-label={productName}
+						maxHeight="30px"
+						maxWidth="150px"
+						hasShadow={false}
+					/>
+				</Button>
+			</Link>
+		</div>
+	);
+
+	const closeButton = (
+		<Button
+			className="menu"
+			variant="ghost"
+			radius="none"
+			color="blue"
+			onClick={toggleMenu}
+			aria-hidden={!isMenuButtonVisible}
+			tabIndex={isMenuButtonVisible ? 0 : -1}
+			aria-label={t`Menu`}
+			aria-haspopup="true"
+			aria-expanded={props.isOpen}
+			aria-controls="main-menu"
+		>
+			{props.isOpen ? <CloseIcon /> : <MenuIcon />}
+		</Button>
+	);
+
+	const { isOpen, className, clientSideConfiguration } = props;
+	const claim = translatedStringFromObject(
+		clientSideConfiguration?.textContent?.product?.claim,
+	);
+
+	const classList = [
+		className,
+		isOpen || !isMenuButtonVisible ? "is-open" : null,
+		"main-menu",
+	].filter(Boolean);
+
+	const { containerRef } = useFocusTrap({
+		disabled: !isMenuButtonVisible || !isOpen,
+	});
+
+	return (
+    <Card asChild variant="surface">
+      <StyledNav
+        className={classList.join(" ")}
+        ref={containerRef as React.RefObject<HTMLElement>}
+      >
+        {homeLink}
+
+        <div className="claim">{claim}</div>
+
+        <div id="main-menu" role="menu">
+          <AppLinks />
+        </div>
+
+        {closeButton}
+      </StyledNav>
+    </Card>
+	);
```

Comment:
**Non-blocker:** Note: I don't want to open a whole new discussion about tabs versus spaces.  I have a very hard time reading this diff and figuring out what has actually changed here. What do you @Mayaryin @opyh think of setting the biome configuration for indentation back from to tabs to spaces for now in order to keep our diffs small with less noise? If we decide to go with tabs we can still do it later in a separate PR and then let biome fix all the files at once. I think it could speed up PR reviews if we postpone this formatting change.